src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -1148,3 +1169,241 @@ const isMobileFieldUpdateAllowed = (
+): ResultAsync<
+  number,
+  SecretsManagerError | SecretsManagerNotFoundError | TwilioCacheError
+> => {
+  const twilioCredentialsData: TwilioCredentialsData =
+    new TwilioCredentialsData(twilioCredentials)
+
+  const body: PutSecretValueRequest = {
+    SecretId: msgSrvcName,
+    SecretString: twilioCredentialsData.toString(),
+  }
+
+  return ResultAsync.fromPromise(
+    secretsManager.getSecretValue({ SecretId: msgSrvcName }).promise(),
+    (error) => {
+      logger.error({
+        message: 'Twilio Credentials do not exist in Secrets Manager',
+        meta: {
+          action: 'updateTwilioCredentials',
+          msgSrvcName,
+        },
+        error,
+      })
+
+      return new SecretsManagerNotFoundError(
+        'Twilio Credentials do not exist in Secrets Manager',
+      )
+    },
+  )
+    .andThen(() => {
+      logger.info({
+        message: 'Twilio Credentials has been found in Secrets Manager',
+        meta: {
+          action: 'updateTwilioCredentials',
+          msgSrvcName,
+        },
+      })
+
+      return ResultAsync.fromPromise(
+        secretsManager.putSecretValue(body).promise(),
+        (error) => {
+          logger.error({
+            message: 'Error occurred when updating Twilio in Secret Manager!',
+            meta: {
+              action: 'updateTwilioCredentials',
+              body,
+            },
+            error,
+          })
+
+          return new SecretsManagerError(
+            'Error occurred when updating Twilio in Secret Manager!',
+          )
+        },
+      )
+    })
+    .andThen(() => {
+      return ResultAsync.fromPromise(
+        Promise.resolve(twilioClientCache.del(msgSrvcName)),
+        // Currently, a call to get twilio credentials will cache the credentials in the twilioCache for ~10s
+        // If a call to retrieve twilio credentials occurs before 10s passes, it will be a cache hit, retrieving
+        // the wrong credentials. Hence we need to clear the cache entry
+
+        (error) => {
+          logger.error({
+            message: 'Error occurred when clearing cache in Secret Manager!',
+            meta: {
+              action: 'updateTwilioCredentials',
+              body,
+            },
+            error,
+          })
+
+          return new TwilioCacheError(
+            'Error occurred when clearing cache entry!',
+          )
+        },
+      )
+    })
+}
+
+export const deleteTwilioCredentials = (
+  formId: string,
+  msgSrvcName: string,
+): ResultAsync<unknown, SecretsManagerError | TwilioCacheError> => {
+  const body: DeleteSecretRequest = {
+    SecretId: msgSrvcName,
+    ForceDeleteWithoutRecovery: true,
+    // Need ForceDeleteWithoutRecovery boolean flag for it to be deleted immediately
+    // If not, the key-value pair will remain in SecretsManager for another 14 days before
+    // being dleted: https://docs.aws.amazon.com/secretsmanager/latest/userguide/manage_delete-secret.html
```

Comment:
is that a bad thing? we can let AWS delete it after 14 days

Reply:
I think it will still exist inside AWS which could lead to some problems, such as not being able to add a new Twilio credential immediately.

When deleting twilio credentials operation is performed, `msgSrvcName` is removed from the document, AWS Secrets Manager is set to delete after 14 days. During this 14 days, users will set the /10,000 free SMSs, if they try to add Twilio Credentials, the server will get an error saying it already exists inside AWS.

Reply:
Is there a way to signal to secrets manager to upsert the data if a to-be-deleted secret is updated?

Reply:
Currently, there dosen't seem to be a way. The only way I can see it being done now is using the [Restore Secret](https://docs.aws.amazon.com/secretsmanager/latest/userguide/manage_restore-secret.html) operation and retry the create opreration.

What we will need to do is to catch the `ResourceExistsException` during create  operation, restore the secret then retry the whole operation.

But I also see there being a problem  with this which is `ResourceExistsException` for a key that is not being scheduled for deletion (key collision, even though unlikely and not suppsoed to happen)

Reply:
I guess it's fine to force delete, maybe add what you typed here into the code comment too for context!